### DISCUSSION TITLE
Colab Playbook Customization Support

### DISCUSSION THREAD

**Discussion by darth2602** [lllyasviel/Fooocus#3056] [Q&A] [answered]
hello I found an alternative version of Foocus  (COLAB)
whose script is basically the same as yours, but two lines have been added:  

1) 
!wget -O /content/Fooocus/models/upscale_models/fooocus_upscaler_s409985e5.bin https://huggingface.co/lllyasviel/misc/resolve/main/fooocus_upscaler_s409985e5.bin?download=true

2)
!wget -O /content/Fooocus/models/checkpoints/realism_engine_sdxl.safetensors https://civitai.com/api/download/models/258380?type=Model&format=SafeTensor&size=pruned&fp=fp16

What's line 1 for? (preloading upscale models?)

Line 2 works as is, but as soon as you modify it to download another model from civitai, you get an error message saying that you need an api .... (????) 

ideas ?

**Comment by mashb1t** [comment]
Hey @darth2602,
1) is for preloading the upscaling model
2) is for downloading realism_engine_sdxl,  i'd suggest to rather use a preset here, but this also works

If other model URLs won't work either Civitai was down again or the creator has set the model access to logged / authenticated users only. If you send a link of a model which isn't possible to download i can double-check.

**Comment by mrboshek** [comment]
This might help, you need to go to Civitai "account settings". This takes you to "Manage Account".  At bottom of page is "Add API key". Copy your key (I think it's 30 characters).
I think you should have something like below, but the x's are replaced by your api key. Right click civitai download button, copy link address, then add ?token=your api key
https://civitai.com/api/download/models/258380?token=xxxxxxxxxxxxxxxxxxxxxxxxxxx

**Comment by darth2602** [comment]
it's GitHub API key ? how to create a key ?

**Reply by mashb1t** [reply, marked as answer]
=> Civitai API key, description: see comment above.

**Reply by darth2602** [reply]
ho....yes ...sorry

**Comment by darth2602** [comment]
argh ! 

!wget -O /content/Fooocus/models/checkpoints/cyberrealisticXL_v11VAE.safetensors https://civitai.com/api/download/models/395041?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

>

--2024-06-01 07:54:13--  https://civitai.com/api/download/models/395041?type=Model
Resolving civitai.com (civitai.com)... 104.22.19.237, 172.67.12.143, 104.22.18.237, ...
Connecting to civitai.com (civitai.com)|104.22.19.237|:443... connected.
HTTP request sent, awaiting response... 401 Unauthorized

Username/Password Authentication Failed.

**Reply by mashb1t** [reply]
just to make sure, you've replaced the xxxx with your token, right? ^^

**Reply by darth2602** [reply]
yes ....^_^

**Reply by darth2602** [reply]
Well.... this one works

!wget -O /content/Fooocus/models/checkpoints/realvisxlV40_V40LightningBakedvae.safetensors https://civitai.com/api/download/models/361593?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx

and the one not

!wget -O /content/Fooocus/models/checkpoints/cyberrealisticXL_v11VAE.safetensors https://civitai.com/api/download/models/395041?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx


I made the two links following the same method (i copy/paste the link from the Civitai model page) the and they can be downloaded without any problem from the browser...

Are there, for example, authorization levels in Civitai ?

**Reply by Rogala** [reply]
If you have generated your API key and I hope saved, the steps to get the correct link are very simple.

Go to the page of the model, lora or anything on https://civitai.com/ select the version and copy the path to the download to the photo from the File section, right-click on the word Download-Copy link address.

The version is usually at the top
![image](https://github.com/lllyasviel/Fooocus/assets/7437582/2ad7bca0-30b1-461b-a1bf-af21520da41b)

The 'Files' section on the right below the description
![image](https://github.com/lllyasviel/Fooocus/assets/7437582/1aaa8fd9-065d-4170-9b94-8ba67c687076)

After receiving the link in any text editor modify it to the form, the address you copied+&token=YourAPIkey

`https://civitai.com/api/download/models/395041?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=APIkeys`

**Reply by darth2602** [reply]
YES .... 

HTTP request sent, awaiting response... 401 Unauthorized

Username/Password Authentication Failed.
Already up-to-date

**Reply by darth2602** [reply]
In fact, after a few tries, it works for other models, but is blocked with cyberrealisticXL

**Reply by Rogala** [reply]
Try mine. I don't have a paid account on my 
https://civitai.com/, I'm just wondering why it's not working for you.

https://civitai.com/api/download/models/395041?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=69e0ce2d09de01ab46cb2ad469fda42c

**Comment by mrboshek** [comment]
What Colab are you using. I tried cyberrealisticXL_v11VAE yesterday and it worked.  I keep a list of models and Lora as text file, then  I copy and paste in the notebook.  I use a different format though:
!curl -Lo /content/Fooocus/models/checkpoints/cyberrealistic11.safetensors https://civitai.com/api/download/models/395041?token=place_api_here

**Reply by darth2602** [reply]
!pip install pygit2==1.12.2
%cd /content
!git clone https://github.com/lllyasviel/Fooocus.git
%cd /content/Fooocus

!wget -O /content/Fooocus/models/upscale_models/fooocus_upscaler_s409985e5.bin https://huggingface.co/lllyasviel/misc/resolve/main/fooocus_upscaler_s409985e5.bin?download=true


!wget -O /content/Fooocus/models/checkpoints/cyberrealisticXL_v11VAE.safetensors https://civitai.com/api/download/models/395041?type=Model&format=SafeTensor&size=pruned&fp=fp16&token=xxxxxxxxxxxxxxxxxxxxxxxxxx


!python entry_with_update.py --preset realistic --share --always-high-vram

**Reply by darth2602** [reply]
it' working without :  "format=SafeTensor&size=pruned&fp=fp"

**Reply by darth2602** [reply]
I want to download models from my google drive  , how i can do this ?

for exemple "toto.safetensor" from a "SdModels" folder at the root of the drive
now i'm am here (https://drive.google.com/drive/....  is not correct) 

!wget -O /content/Fooocus/models/checkpoints/toto.safetensors https://drive.google.com/drive/home/content/gdrive/MyDrive/SdModels/toto.safetensors

**Reply by Rogala** [reply]
![image](https://github.com/lllyasviel/Fooocus/assets/7437582/c1601ad3-5e9e-48d6-80ab-8c399925ecd7)
Connect the disk directly and specify the paths in the config.txt file

**Reply by darth2602** [reply]
?

**Reply by Rogala** [reply]
The point is that you can connect google disk in the colab environment as a folder and by changing the paths in the config.txt file, do not load models and Lora, but take them directly from the folder

**Reply by darth2602** [reply]
hum....ok

**Reply by darth2602** [reply]
is it possible to access to "https://drive.google.com/drive/home/content/foocus"   with filezilla pro ?  (google drive protocol) 
direct download from civitai is very slow, and the import from my local SSD or google drive not working for me
